File: src/rules/hallucinated-apis.py
import jwt
import bcrypt


def check_token(token):
    # ruleid: clipeus-hallucinated-security-api-python
    return jwt.validate(token)


def check_password(pw, hashed):
    # ruleid: clipeus-hallucinated-security-api-python
    return bcrypt.verify(pw, hashed)


def check_password_real(pw, hashed):
    # ok: clipeus-hallucinated-security-api-python
    return bcrypt.checkpw(pw, hashed)

File: src/rules/hallucinated-apis.js
const jwt = require('jsonwebtoken');
const bcrypt = require('bcrypt');
const crypto = require('crypto');

function checkToken(token) {
  // ruleid: patronus-hallucinated-security-api-js
  return jwt.validate(token);
}

function checkPassword(pw, hash) {
  // ruleid: patronus-hallucinated-security-api-js
  return bcrypt.verify(pw, hash);
}

function encryptData(data, key) {
  // ruleid: patronus-hallucinated-security-api-js
  return crypto.encrypt(data, key);
}

function checkPasswordReal(pw, hash) {
  // ok: patronus-hallucinated-security-api-js
  return bcrypt.compare(pw, hash);
}

function verifyTokenReal(token, key) {
  // ok: patronus-hallucinated-security-api-js
  return jwt.verify(token, key);
}
